pkg/channel/consolidated/status/status.go

```diff
@@ -182,9 +191,26 @@ func (m *Prober) IsReady(ctx context.Context, ch messagingv1beta1.KafkaChannel,
 				state.lastAccessed = time.Now()
 				logger.Debugw("Subscription is cached. Checking readiness",
 					zap.Any("subscription", sub.UID))
-				return m.checkReadiness(state), true
+				if state.ready {
+					/* This subscription became ready once, that's sufficient for us for now.
+					Ideally we'd have a way to know if each partition already have at least one offset commit, but that's
+					another problem for another day
+					*/
+					return true, true
+				} else {
+					if target.PodIPs.Len() == state.probedPods.Len() && target.PodIPs.HasAll(state.probedPods.List()...) {
+						return false, true
+					} else {
+						logger.Debugw("Dispatcher pods changed. Canceling old probes and restarting",
+							zap.Any("subscription", sub.UID))
+						// the dispatcher pods changed, we need to cancel old probing and start again
+						m.mu.Unlock()
```

Comment:
why're we doing the unlocking, cancelprobing, re-lock when it seems like CancelProbing does the same thing as lines 214-216?

I'd prefer an approach where we're not unlocking, calling another function and then relocking like this tbh